### PR TITLE
add functionality for continuous fine-tuning on subpopulation

### PR DESCRIPTION
**Pull request by kirilklein**
## Summary by CodeRabbit

* **New Features**
  * Subpopulation-specific causal finetuning with bootstrap fold creation and continuation support
  * Azure ML pipeline and component to run subpopulation finetune → calibrate → estimate (handles optional inputs)
  * Runtime entrypoint script and YAML config for subpopulation finetuning
  * Trainer option to freeze the encoder at initialization
  * Dataset utility to drop constant outcome labels

* **Tests**
  * Tests for fold creation/validation and encoder-freeze-at-init behavior

* **Chores**
  * Optional pretrain check in setup and minor CI workflow formatting tweaks

### CODE REVIEW

**Comment by kirilklein** [comment]
@Montgomeryyyy we can skip calibration for outcomes entirely, only ps needs to be calibrated. I can implement this quickly. Depending how many jobs you have run already, it could just be a separate script

**Comment by Montgomeryyyy** [comment]
> @Montgomeryyyy we can skip calibration for outcomes entirely, only ps needs to be calibrated. I can implement this quickly. Depending how many jobs you have run already, it could just be a separate script

I have mostly run the cvd and diab sub cohorts, and this does not take long. So if you can implement if quickly that would be nice:)
